.github/workflows/build-and-release.yml

```diff
@@ -86,17 +138,71 @@ jobs:
       RUNNER_LABEL: ${{ needs.start-runner.outputs.label }}
       INFRASTRUCTURE_FOLDER: "infrastructure/aws"
       PUBLIC_INFRA_DEPLOYMENT: true
-      ENVIRONMENT: dev
+      ENVIRONMENT: ${{ needs.set-vars.outputs.environment }}
       INFRA_CONFIG_REPO: ${{ needs.set-vars.outputs.infra-config-repo }}
-      CORE_INFRA_REF: ${{ github.event.inputs.core-repo-reference }}
-      INFRA_CONFIG_REF: ${{ github.event.inputs.config-repo-reference }}
-      IMAGE_TAG: ${{ github.event.inputs.commit-hash }}
+      CORE_INFRA_REF: ${{ needs.set-vars.outputs.core-repo-reference }}
+      INFRA_CONFIG_REF: ${{ needs.set-vars.outputs.config-repo-reference }}
+      IMAGE_TAG: ${{ needs.set-vars.outputs.commit-hash }}
     secrets:
       AWS_GITHUBRUNNER_PAT: ${{ secrets.AWS_GITHUBRUNNER_PAT}}
       AWS_GITHUBRUNNER_PAT_USER: ${{ secrets.AWS_GITHUBRUNNER_PAT_USER }}
       AWS_REGION: ${{ secrets.AWS_REGION }}
       AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID }}
 
+  notify-slack:
```

Comment:
Awesome! I love a good slack notification. We should look at turning this into a reusable GH action for sure, would be useful on pretty much every project I imagine. I can raise a ticket if you haven't already and we can bake it into the cookiecutter.